pkgs/tools/security/ghidra/layers/1_util.nix

```diff
@@ -0,0 +1,63 @@
+self: super: {
+    config = {
+      pkg_path = "lib/ghidra";
+      ghidraDevPath = self.config.pkg_path + "/Extensions/Eclipse/GhidraDev/GhidraDev-2.0.0.zip";
+      ghidraDevVersion = "2.0.0";
+      launchers = [
+          "ghidraRun" "support/analyzeHeadless" "support/buildGhidraJar"
+          "support/convertStorage" "support/dumpGhidraThreads" "support/ghidraDebug"
+          "support/pythonRun" "support/sleigh"
+          ];
+      defaultOpts = import ../lib/defaultOpts.nix;
+      };
+
+
+    lib = {
+      # nix-shell -p nix-prefetch-github --run "nix-prefetch-github owner repo > ./plugins/json/theplugin.json"
+      fetchGitHubJSON = {JSONfile, ...}@args:
+        self.pkgs.fetchFromGitHub ({ inherit (self.pkgs.lib.importJSON JSONfile) owner repo rev sha256; } // (
+          builtins.removeAttrs args [ "JSONfile" ]
+          ));
+
+      # { args ? defaultOpts.args, debug ? defaultOpts.debug }: # The signature
+      mkRunline = self.callPackage ((import ../lib/mkRunline.nix) self.config.defaultOpts) {};
+
+      poorMkGradle = self.callPackage ../lib/poorMkGradle.nix {};
+
+      jdkWrapper = src: dst: ''
+        makeWrapper "$out/${src}" "$out/${dst}" \
+          --prefix PATH : '${self.pkgs.lib.makeBinPath [ self.jdk ]}' \
+          --run '. ${self.jdk}/nix-support/setup-hook' #set JAVA_HOME
```

Comment:
We not setting JAVA_HOME directly?

Reply:
I wasn't really sure what to do about this. It makes some kind of sense to me to delegate setting whatever environment stuff a dependency implies, by relying on the dependency's scripts?